tests/mod.rs

```diff
@@ -197,35 +197,35 @@ fn should_work_for_indexed() {
 }
 
 #[test]
-#[should_panic(expected = "Error decoding field Indexed.0")]
+#[should_panic(expected = "Not enough data to fill buffer")]
 fn correct_error_for_indexed_0() {
 	let mut wrong: &[u8] = b"\x08";
 	Indexed::decode(&mut wrong).unwrap();
 }
 
 #[test]
-#[should_panic(expected = "Error decoding field Indexed.1")]
+#[should_panic(expected = "Not enough data to fill buffer")]
 fn correct_error_for_indexed_1() {
 	let mut wrong: &[u8] = b"\0\0\0\0\x01";
 	Indexed::decode(&mut wrong).unwrap();
 }
 
 #[test]
-#[should_panic(expected = "Error decoding field EnumType :: B.0")]
+#[should_panic(expected = "Not enough data to fill buffer")]
 fn correct_error_for_enumtype() {
 	let mut wrong: &[u8] = b"\x01";
 	EnumType::decode(&mut wrong).unwrap();
 }
 
 #[test]
-#[should_panic(expected = "Error decoding field Struct.a")]
+#[should_panic(expected = "Not enough data to fill buffer")]
 fn correct_error_for_named_struct_1() {
 	let mut wrong: &[u8] = b"\x01";
 	Struct::<u32, u32, u32>::decode(&mut wrong).unwrap();
 }
 
 #[test]
-#[should_panic(expected = "Error decoding field Struct.b")]
+#[should_panic(expected = "Not enough data to fill buffer")]
```

Comment:
Seeing this, shouldn't we implement `Debug` as `Display`?

Reply:
actually the error returned is:
```
thread 'correct_error_for_indexed_0' panicked at 'called `Result::unwrap()` on an `Err` value: Error { cause: Some(Error { cause: None, desc: "Not enough data to fill buffer" }), desc: "Could not decode `Indexed::0`" }', tests/mod.rs:202:33
```

Should_panic just match for the inner string

Reply:
so I don't know what is better ?

Reply:
Ahh yeah, no that looks good.

The error you have shown there is without chaining or?

Reply:
there is no longer "std" without chaining (as std now add feature chaining). But yes `"Not enough data to fill buffer"` is the root cause.
(And the error shown in https://github.com/paritytech/parity-scale-codec/pull/242#discussion_r563741838 is with chaining)

Reply:
note that we don't run test for chaining without std, should we do ?
we do run `cargo check`.

Reply:
> note that we don't run test for chaining without std, should we do ?
> we do run `cargo check`.

That should be fine.

What I'm more wondering:
> `thread 'correct_error_for_indexed_0' panicked at 'called `Result::unwrap()` on an `Err` value: Error { cause: Some(Error { cause: None, desc: "Not enough data to fill buffer" }), desc: "Could not decode `Indexed::0`" }', tests/mod.rs:202:33`

Shouldn't this mention the `Struct`?

Reply:
Saying that we could not decode `Struct.a`?

Reply:
the struct is named `Indexed`.
`"Indexed::0"` means the first unnamed type of the struct `Indexed`, maybe we should give more explicit error message

Reply:
Yeah I know this, but below we decode the `Struct`. Or did you just pasted the wrong error here? :D

Reply:
I see that we decode `Indexed` here: correct_error_for_indexed_0